syntax = "proto3";
package chat;

// 비어있는 메시지 정의
message Empty {}

// 메시지 구조 정의
message ChatMessage {
    string user = 1;    // 사용자 이름
    string message = 2; // 전송할 메시지
}

// gRPC 서비스 정의
service ChatService {
    rpc Chat(stream ChatMessage) returns (Empty); // 클라이언트 측 스트리밍
    rpc broadcast_message(Empty) returns (stream ChatMessage); // 서버 측 스트리밍
}